### DISCUSSION TITLE
Usage of official mongo driver with actix-web

### DISCUSSION THREAD

**Discussion by maku** [actix/actix-web#1810] [Q&A]
I am new with rust and would like to use the official mongo driver with actix-web. But I really do not know how.
The default of the mongo driver ist async which makes things not easier.

So it would be great if anybody can give me an example how to do this....

**Comment by fakeshadow** [comment]
mongo drivers default to tokio 0.2 runtime which is supported directly by actix-web. 
You can check the readme of https://github.com/mongodb/mongo-rust-driver on how to use the async APIs.

**Reply by maku** [reply]
Yeah, but I am not really experienced in rust and actix-web:

So in my main I have something like that:

```
let client_options = ClientOptions::parse("mongodb://localhost:27017").await;
let db: Database = Client::with_options(client_options.unwrap()).unwrap().database("mydatabase");

HttpServer::new(move || {
        App::new()
            .data(|| db.clone())
            .route("/status", web::get().to(status))
    ...
 })
 .bind("127.0.0.1:7777")?
.run()
.await

```
I do not know if this correct to be able to consume "db" in my "status" function.

What I need would be an example how the function ("status") has to look like to be able to use db from "web::Data"

Something like
`pub async fn status(state: web::Data<Database>) ...`

**Comment by robjtede** [comment]
Check out this example for a general guide: https://github.com/actix/examples/blob/master/async_db/src/main.rs.

```rust
App::new()
  .data(db.clone())

...

pub async fn status(state: web::Data<Database>) { ... }
```

**Reply by maku** [reply]
@robjtede thanks for answering, I looked at this already. But as far as I know the mongo driver has a build in pool management.
